src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3485,6 +3485,92 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
+        {
+            var partsBuilder = ArrayBuilder<IInterpolatedStringContentOperation>.GetInstance(operation.Parts.Length);
+            ArrayBuilder<IOperation> formatBuilderOpt = null;
+            foreach (IInterpolatedStringContentOperation element in operation.Parts)
+            {
+                IInterpolatedStringContentOperation rewrittenElement;
+                if (element.Kind == OperationKind.Interpolation)
+                {
+                    var interpolation = (IInterpolationOperation)element;
+                    _evalStack.Push(Visit(interpolation.Expression));
+
+                    if (interpolation.Alignment != null)
+                    {
+                        _evalStack.Push(Visit(interpolation.Alignment));
+                    }
+
+                    if (interpolation.FormatString != null)
+                    {
+                        Debug.Assert(interpolation.FormatString.Kind == OperationKind.Literal);
+                        var rewrittenFormatString = VisitLiteral((ILiteralOperation)interpolation.FormatString, captureIdForResult);
+                        formatBuilderOpt = formatBuilderOpt ?? ArrayBuilder<IOperation>.GetInstance(operation.Parts.Length);
+                        formatBuilderOpt.Add(Visit(interpolation.FormatString));
+                    }
+
+                    // We generate the rewritten IInterpolationOperation after visiting all interpolations.
+                    // Add null as a placeholder.
+                    rewrittenElement = null;
+                }
+                else
+                {
+                    var interpolatedStringText = (IInterpolatedStringTextOperation)element;
+                    Debug.Assert(interpolatedStringText.Text.Kind == OperationKind.Literal);
+                    var rewrittenInterpolationText = VisitLiteral((ILiteralOperation)interpolatedStringText.Text, captureIdForResult);
+                    rewrittenElement = new InterpolatedStringText(rewrittenInterpolationText, semanticModel: null, element.Syntax, element.Type, element.ConstantValue, element.IsImplicit);
+                }
+
+                partsBuilder.Add(rewrittenElement);
+            }
+
+            for (int i = operation.Parts.Length - 1; i >= 0; i--)
+            {
+                IInterpolatedStringContentOperation element = operation.Parts[i];
+                if (element.Kind == OperationKind.Interpolation)
+                {
+                    Debug.Assert(partsBuilder[i] == null);
+
+                    var interpolation = (IInterpolationOperation)element;
+
+                    IOperation rewrittenFormatString;
+                    if (interpolation.FormatString != null)
+                    {
+                        Debug.Assert(interpolation.FormatString.Kind == OperationKind.Literal);
+                        rewrittenFormatString = formatBuilderOpt[formatBuilderOpt.Count - 1];
```

Comment:
>formatBuilderOpt[formatBuilderOpt.Count - 1]; [](start = 48, length = 45)

I think we have a ```Pop``` helper method that will also take care of changing the size of the builder.